metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizerData.java

```diff
@@ -397,18 +387,34 @@ private MatchingAclRule findAclRule(
         return matchingAclBuilder.build();
     }
 
+    /**
+     * Use a binary search to find the index of the first ACL which is greater than or
+     * equal to the given ACL. This may be equal to the end of the array if there are
+     * no such ACLs.
+     */
+    private int indexOfFirstAclGreaterThanOrEqualTo(StandardAcl exemplar) {
+        int i = Arrays.binarySearch(acls,
+                new StandardAclWithId(Uuid.ZERO_UUID, exemplar),
+                StandardAclWithId.ACL_COMPARATOR);
+        // Arrays.binarySearch returns a positive number if it found an exact match, and
+        // a negative number otherwise.
```

Comment:
Might comment about what the negative return value indicates. It helps L404 make more sense :)